src/backend/env/mod.rs

```diff
@@ -1215,12 +1223,22 @@ impl State {
     }
 
     pub fn collect_new_rewards(&mut self) -> HashMap<UserId, u64> {
-        self.users
-            .values_mut()
-            .filter_map(|u| {
-                (u.rewards() > 0).then_some((u.id, u.take_positive_rewards() as Credits))
-            })
-            .collect()
+        let mut payouts = HashMap::default();
+
+        for user in self.users.values_mut() {
+            let rewards = user.take_positive_rewards();
+            if rewards == 0 {
+                continue;
+            };
+            // All miner rewards are burned.
+            if user.miner {
+                self.burned_cycles += rewards;
```

Comment:
Just to double check: this will account these rewards as revenue, so they will be distributed to users as revenue?

Reply:
Yes!